[PE-D][Tester B] User behavior for `grpadd` is not handled on duplicate member names with different casing

## Description 
When `grpadd` commands receives arguments with the same name but different casing (e.g., `/n Alice /n alice`), HitList just stucked there and users are unable to enter the command. 

## Expected behavior
The system should consistently treat names as case-insensitive during the grpadd process. It should either:
1. Automatically deduplicate the names before processing.
2. Display a validation error message to the user 

To ensure that no more than one contact with the same name (regardless of casing) can be added to the group.

## Actual behavior
No error message. 

## Step to reproduce
0. `clear`: Clear HitList
1. `add /n Alice /p 111` : Add a contact for Alice
2. `grpadd /g Team /n Alice /n alice` --> Observe that legitimate user behavior is not handled



-------------
Labels: `severity.Medium` `type.FunctionalityBug`

Important bug.

Stack Trace
```
INFO: ----------------[USER COMMAND][grpadd /g Team /n Alice /n alice]
Exception in thread "JavaFX Application Thread" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1857)
	...
Caused by: hitlist.model.person.exceptions.DuplicatePersonException: Operation would result in duplicate persons
	at hitlist.model.person.UniquePersonList.add(UniquePersonList.java:46)
	at hitlist.model.group.Group.addMember(Group.java:38)
	at hitlist.logic.commands.AddGroupCommand.execute(AddGroupCommand.java:74)
	at hitlist.logic.LogicManager.execute(LogicManager.java:54)
	at hitlist.ui.MainWindow.executeCommand(MainWindow.java:274)
	at hitlist.ui.CommandBox.handleCommandEntered(CommandBox.java:45)
	... 58 more
Caused by: java.lang.reflect.InvocationTargetException

Caused by: hitlist.model.person.exceptions.DuplicatePersonException: Operation would result in duplicate persons
```

Caused by snippet:

<img width="1500" height="843" alt="Image" src="https://github.com/user-attachments/assets/5b50e797-7791-4342-abae-8de491bce41e" />

# Suggestions

- Since memberNames should be unique anyways, deduplicate before running the for-loop
- For the sake of UX, we should throw an error if there are duplicate names provided
- Either we check at the string-level (`alice` and `Alice` in `memberNames` = ERROR)
- Or we check at the object-level
    - Query for `alice` -> `"hitlist.model.person.Person{name=Alice, phone=111, email=Optional.empty, address=Optional.empty}"`
    - Query for `Alice` -> `"hitlist.model.person.Person{name=Alice, phone=111, email=Optional.empty, address=Optional.empty}"`

--- a/src/main/java/hitlist/model/person/Name.java
+++ b/src/main/java/hitlist/model/person/Name.java
@@ -13,7 +13,7 @@ public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
             "Names should only contain alphabetic characters, dashes,"
-                    + " apostrophes, and spaces, and it should not be blank";
+             + " apostrophes, and spaces, and it should not be blank";
 
     /*
      * Names should only contain alphabetic characters, spaces, -, and ', and it should not be blank

--- a/src/main/java/hitlist/model/person/Name.java
+++ b/src/main/java/hitlist/model/person/Name.java
@@ -13,7 +13,7 @@ public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
             "Names should only contain alphabetic characters, dashes,"
-             + " apostrophes, and spaces, and it should not be blank";
+            + " apostrophes, and spaces, and it should not be blank";
 
     /*
      * Names should only contain alphabetic characters, spaces, -, and ', and it should not be blank

--- a/src/main/java/hitlist/model/person/Name.java
+++ b/src/main/java/hitlist/model/person/Name.java
@@ -3,6 +3,8 @@ package hitlist.model.person;
 import static hitlist.commons.util.AppUtil.checkArgument;
 import static java.util.Objects.requireNonNull;
 
+import java.util.Locale;
+
 /**
  * Represents a Person's name in the HitList.
  * Guarantees: immutable; is valid as declared in {@link #isValidName(String)}
@@ -11,7 +13,7 @@ public class Name {
 
     public static final String MESSAGE_CONSTRAINTS =
             "Names should only contain alphabetic characters, dashes,"
-            + " apostrophes, and spaces, and it should not be blank";
+                    + " apostrophes, and spaces, and it should not be blank";
 
     /*
      * Names should only contain alphabetic characters, spaces, -, and ', and it should not be blank
@@ -38,7 +40,6 @@ public class Name {
         return test.matches(VALIDATION_REGEX);
     }
 
-
     @Override
     public String toString() {
         return fullName;
@@ -61,7 +62,6 @@ public class Name {
 
     @Override
     public int hashCode() {
-        return fullName.hashCode();
+        return fullName.toLowerCase(Locale.ROOT).hashCode();
     }
-
 }

--- a/src/test/java/hitlist/logic/parser/AssignGroupCommandParserTest.java
+++ b/src/test/java/hitlist/logic/parser/AssignGroupCommandParserTest.java
@@ -51,9 +51,10 @@ public class AssignGroupCommandParserTest {
     @Test
     public void parse_compulsoryFieldMissing_failure() {
         String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, AssignGroupCommand.MESSAGE_USAGE);
-
         assertParseFailure(parser, NAME_DESC_AMY, expectedMessage);
+
         assertParseFailure(parser, GROUP_NAME_DESC_STUDENTS, expectedMessage);
+
         assertParseFailure(parser, VALID_NAME_AMY + " " + VALID_GROUP_NAME_STUDENTS, expectedMessage);
     }
 
@@ -62,7 +63,6 @@ public class AssignGroupCommandParserTest {
         assertParseFailure(parser, INVALID_NAME_DESC + GROUP_NAME_DESC_STUDENTS, Name.MESSAGE_CONSTRAINTS);
 
         assertParseFailure(parser, NAME_DESC_AMY + INVALID_GROUP_NAME_DESC, GroupName.MESSAGE_CONSTRAINTS);
-
         assertParseFailure(parser, PREAMBLE_NON_EMPTY + NAME_DESC_AMY + GROUP_NAME_DESC_STUDENTS,
                 String.format(MESSAGE_INVALID_COMMAND_FORMAT, AssignGroupCommand.MESSAGE_USAGE));
     }

--- a/src/test/java/hitlist/model/person/NameTest.java
+++ b/src/test/java/hitlist/model/person/NameTest.java
@@ -1,6 +1,7 @@
 package hitlist.model.person;
 
 import static hitlist.testutil.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
@@ -47,6 +48,9 @@ public class NameTest {
         // same values -> returns true
         assertTrue(name.equals(new Name("Valid Name")));
 
+        // same values, different casing -> returns true
+        assertTrue(name.equals(new Name("valid name")));
+
         // same object -> returns true
         assertTrue(name.equals(name));
 
@@ -58,5 +62,13 @@ public class NameTest {
 
         // different values -> returns false
         assertFalse(name.equals(new Name("Other Valid Name")));
+    }
+
+    @Test
+    public void hashCode_sameNameDifferentCasing_sameHashCode() {
+        Name firstName = new Name("Alice");
+        Name secondName = new Name("alice");
+
+        assertEquals(firstName.hashCode(), secondName.hashCode());
     }
 }

